faculty/clients/job.py

```diff
@@ -125,6 +150,73 @@ def make_job_summary(self, data):
         return JobSummary(**data)
 
 
+class InstanceSizeSchema(BaseSchema):
+    milli_cpus = fields.Float(data_key="milliCpus", required=True)
+    memory_mb = fields.Float(data_key="memoryMb", required=True)
+
+    @post_load
+    def make_instance_size(self, data):
+        return InstanceSize(**data)
+
+
+class JobParameterSchema(BaseSchema):
+    name = fields.String(required=True)
+    param_type = EnumField(
+        ParameterType, data_key="type", by_value=True, required=True
+    )
+    default = fields.String(required=True)
+    required = fields.Boolean(required=True)
+
+    @post_load
+    def make_job_parameter(self, data):
+        return JobParameter(**data)
+
+
+class JobCommandSchema(BaseSchema):
+    name = fields.String(required=True)
+    parameters = fields.List(
+        fields.Nested(JobParameterSchema, required=True), required=True
+    )
+
+    @post_load
+    def make_job_command(self, data):
+        return JobCommand(**data)
+
+
+class JobDefinitionSchema(BaseSchema):
+    working_dir = fields.String(data_key="workingDir", required=True)
+    command = fields.Nested(JobCommandSchema, required=True)
+    image_type = fields.String(data_key="imageType", required=True)
+    conda_environment = fields.String(
+        data_key="condaEnvironment", missing=None, required=False
+    )
+    environment_ids = fields.List(
+        fields.String(), data_key="environmentIds", required=True
+    )
+    instance_size_type = fields.String(
```

Comment:
What would you think about adding [schema level validation](https://marshmallow.readthedocs.io/en/3.0/extending.html#schema-level-validation) to check that the instance_size is set when the instance_size_type is 'custom' and not set otherwise?

I think it's important to then test both cases to catch (for example) the fact that the instance_size below is not always required.